src/test/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/KubernetesPipelineRJRTest.java

```diff
@@ -12,6 +19,44 @@ public KubernetesPipelineRJRTest() throws UnknownHostException {
 
     @Test
     public void basicPipeline() throws Throwable {
+        RunId runId = createWorkflowJobThenScheduleRun();
         rjr.runRemotely(new AssertBuildStatusSuccess(runId));
     }
+
+    @Test
+    public void restartDuringPodLaunch() throws Throwable {
+        // try to run something on a pod which is not schedulable (disktype=special)
+        RunId build = rjr.runRemotely(new CreateWorkflowJobThenScheduleTask(
+                KubernetesTestUtil.loadPipelineScript(getClass(), name.getMethodName() + ".groovy")));
+        // the pod is created, but not connected yet
+        rjr.runRemotely(new AssertBuildLogMessage("Created Pod", build));
+        // restart
+        rjr.stopJenkins();
+        rjr.startJenkins();
+        // update k8s to make a node suitable to schedule (add disktype=special to the node)
```

Comment:
A little scary to have the test even temporarily patch the `Node`! Would be safer to do something to the namespace like a quota :thinking: Fine enough for Kind/CI, could just become a problem if we try to run tests on a permanent cluster.

Reply:
This was the easiest way I found, and with the `finally` it's granted to be back to the original state.

Reply:
So long as the JVM terminates normally, yes.